spectrochempy/analysis/fast_ica.py

```diff
@@ -0,0 +1,350 @@
+        help="Number of components (sources) to use.",
+    ).tag(config=True)
+
+    algorithm = tr.Enum(
+        ["parallel", "deflation"],
+        default_value="parallel",
+        help=("""Specify which algorithm to use for FastICA."""),
+    ).tag(config=True)
+
+    whiten = tr.Union(
+        (tr.Enum(["arbitrary-variance", "unit-variance"]), tr.Bool(False)),
+        default_value="arbitrary-variance",
+        allow_none=True,
+        help=(
+            """Specify the whitening strategy to use.
+
+- ``"arbitrary-variance"``\ : a whitening with variance arbitrary is used.
+- "unit-variance" : the whitening matrix is rescaled to ensure that each recovered
+  source has unit variance.
+- False : the data is already considered to be whitened, and no whitening is performed.
+"""
+        ),
+    ).tag(config=True)
+
+    fun = tr.Union(
+        (tr.Enum(["logcosh", "exp", "cube"]), tr.Callable(), tr.Unicode()),
+        default_value="logcosh",
+        allow_none=True,
+        help=(
+            r"""The functional form of the function used in the approximation to neg-entropy.
+
+- ``string``\ : could be either ``"logcosh"``\ , ``"exp"``\ , or ``"cube"``\ .
+- ``callable``\ : You can provide your own function. It should return a tuple containing
+  the value of the function, and of its derivative, in the point. The derivative should
+  be averaged along its last dimension.
+
+    .. code-block::
+
+        def my_g(x):
+            return x ** 3, (3 * x ** 2).mean(axis=-1)
+
+    .. note::
+        ``fun`` can be also a serialized function created using dill and base64
+        python libraries. Normally not used directly, it is here for internal
+        process. """
+        ),
+    ).tag(config=True)
+
+    fun_args = tr.Dict(
+        default_value=None,
+        allow_none=True,
+        help=(
+            """Arguments to send to the functional form.
+
+If empty or None and if :code:`fun=="logcosh"`` , `fun_args` will take value
+:code:`{alpha : 1.0}`."""
+        ),
+    ).tag(config=True)
+
+    tol = tr.Float(
+        default_value=1e-4,
+        help="Tolerance at which the un-mixing matrix is considered to have converged.",
+    ).tag(config=True)
+
+    max_iter = tr.Integer(
+        default_value=200, help="Maximum number of iterations before timing out."
+    ).tag(config=True)
+
+    w_init = tr.Union(
+        (NDDatasetType(),),
+        default_value=None,
+        allow_none=True,
+        help=(
+            r"""Initial un-mixing array.
+
+NDDataset or array-like of shape (n_components, n_components). If w_init=None, then an
+array of values drawn from a normal distribution is used."""
+        ),
+    ).tag(config=True)
+
+    whiten_solver = tr.Enum(
+        ["svd", "eigh"],
+        default_value="svd",
+        help=(
+            """The solver to use for whitening.
+
+- ``"svd"``\ : is more stable numerically if the problem is degenerate, and often faster
+  when :code:`n_observations <= n_features`\ .
```

Comment:
here I would write (but this is really a detail):

    when :term:`n_observations` <= :term:`n_features`\ .
    - ``"eigh"``\ : is generally more memory efficient when
     :term:`n_observations` >= :term:`n_features`\ , and can be faster when
     :term:`n_observations` >= 50 * :term:`n_features`\ . """